src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,241 @@
+import React, { FC, useLayoutEffect, useRef, useState } from 'react';
+
+import { mergeClasses } from '../../shared/utilities';
+import styles from './slider.module.scss';
+import { SliderMarker, SliderProps } from './Slider.types';
+
+const thumbDiameter: number = +styles.thumbDiameter;
+const markerWidth: number = +styles.markerWidth;
+
+function asc(a: number, b: number) {
+    return a - b;
+}
+
+export function valueToPercent(value: number, min: number, max: number) {
+    return ((value - min) * 100) / (max - min);
+}
+
+export const Slider: FC<SliderProps> = ({
+    ariaLabel,
+    autoFocus = false,
+    className,
+    disabled = false,
+    id,
+    min = 0,
+    max = 100,
+    name,
+    onChange,
+    showLabels = true,
+    showMarkers = false,
+    step = 1,
+    value,
+}) => {
+    const isRange = Array.isArray(value);
+    const [values, setValues] = useState(isRange ? value.sort(asc) : [value]);
+    const railRef = useRef<HTMLDivElement>(null);
+    const lowerLabelRef = useRef<HTMLInputElement>(null);
+    const upperLabelRef = useRef<HTMLInputElement>(null);
+    const trackRef = useRef<HTMLDivElement>(null);
+    let [showMinLabel, setShowMinLabel] = useState(showLabels);
+    let [showMaxLabel, setShowMaxLabel] = useState(showLabels);
+    let [markers, setMarkers] = useState<SliderMarker[]>([]);
+
+    function getIdentifier(baseString: string, index: number): string {
+        if (!baseString) {
+            return '';
+        }
+        const idTokens: Array<string | number> = [baseString];
+        if (isRange) {
+            idTokens.push(index);
+        }
+        return idTokens.join('-');
+    }
+
+    function handleChange(newVal: number, index: number) {
+        const newValues = [...values];
+        newValues.splice(index, 1, newVal);
+        newValues.sort(asc);
+        setValues(newValues);
+        onChange?.(isRange ? [...newValues] : newValues[0]);
```

Comment:
Is there potential for a race condition between the state and change events? If so, please consider use of async.